apps/files_sharing/lib/sharedstorage.php

```
@@ -78,15 +85,19 @@ private function init() {
 			return;
 		}
 		$this->initialized = true;
-		Filesystem::initMountPoints($this->share['uid_owner']);
-		$sourcePath = $this->ownerView->getPath($this->share['file_source']);
-		list($this->sourceStorage, $sourceInternalPath) = $this->ownerView->resolvePath($sourcePath);
-		$this->sourceRootInfo = $this->sourceStorage->getCache()->get($sourceInternalPath);
+		try {
+			Filesystem::initMountPoints($this->share['uid_owner']);
+			$sourcePath = $this->ownerView->getPath($this->share['file_source']);
+			list($this->sourceStorage, $sourceInternalPath) = $this->ownerView->resolvePath($sourcePath);
+			$this->sourceRootInfo = $this->sourceStorage->getCache()->get($sourceInternalPath);
+		} catch (\Exception $e) {
+			$this->logger->logException($e);
```

The trouble with this is that the code will continue running and will continue using the unitialized SharedStorage, so calling other methods will either fail silently or run into other issues when accessing uninitialized members.

Do we know the root cause of invalid shares reaching this part of the code ?